swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/AbstractSwaggerGenerator.java

```diff
@@ -311,6 +314,21 @@ protected void scanMethods() {
         throw new IllegalStateException(
             String.format("OperationId must be unique. method=%s:%s.", cls.getName(), method.getName()));
       }
+      Map<String, Model> originsDefinitions = swagger.getDefinitions();
+      if (!CollectionUtils.isEmpty(originsDefinitions)) {
+        originsDefinitions.forEach((key, value) -> {
+          if (definitions.containsKey(key) && !value.equals(definitions.get(key))) {
+            throw new IllegalArgumentException(
+                "the parameter typeName '" + key + "' duplicate in '" + this.cls.getName()
+                    + "', need to rename it");
+          }
+        });
+        definitions.putAll(originsDefinitions);
+      }
+      swagger.setDefinitions(null);
```

Comment:
It's a bit complicated. 

Maybe you can have a very simple solution to extend Swagger `public void addDefinition(String key, Model model)` method